# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLParent.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLParent.h
--- a/dom/canvas/WebGLParent.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLParent.h	Wed Jan 08 22:19:16 2020 +0000
@@ -8,12 +8,11 @@
 
 #include "mozilla/GfxMessageUtils.h"
 #include "mozilla/dom/PWebGLParent.h"
-#include "mozilla/dom/WebGLCrossProcessCommandQueue.h"
-#include "mozilla/dom/WebGLErrorQueue.h"
 #include "mozilla/WeakPtr.h"
 
 namespace mozilla {
 
+class HostWebGLCommandSink;
 class HostWebGLContext;
 
 namespace layers {
@@ -22,21 +21,22 @@
 
 namespace dom {
 
-class WebGLParent : public PWebGLParent, public SupportsWeakPtr<WebGLParent> {
+class WebGLParent final : public PWebGLParent,
+                          public SupportsWeakPtr<WebGLParent> {
+  friend PWebGLParent;
+
  public:
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(WebGLParent, override);
   MOZ_DECLARE_WEAKREFERENCE_TYPENAME(WebGLParent)
 
-  static WebGLParent* Create(
-      WebGLVersion aVersion,
-      UniquePtr<mozilla::HostWebGLCommandSink>&& aCommandSink,
-      UniquePtr<mozilla::HostWebGLErrorSource>&& aErrorSource);
+  static RefPtr<WebGLParent> Create(const webgl::InitContextDesc&,
+                                    webgl::InitContextResult* out);
 
   already_AddRefed<layers::SharedSurfaceTextureClient> GetVRFrame();
 
- protected:
-  friend PWebGLParent;
-
-  WebGLParent(UniquePtr<HostWebGLContext>&& aHost);
+ private:
+  WebGLParent();
+  ~WebGLParent();
 
   bool BeginCommandQueueDrain();
   static bool MaybeRunCommandQueue(const WeakPtr<WebGLParent>& weakWebGLParent);